src/tests/detect-file-data.c

```diff
@@ -29,182 +29,6 @@
-                0x6E, 0x3A, 0x20, 0x61, 0x74, 0x74, 0x61, 0x63,
-                0x68, 0x6D, 0x65, 0x6E, 0x74, 0x3B, 0x20, 0x66,
-                0x69, 0x6C, 0x65, 0x6E, 0x61, 0x6D, 0x65, 0x3D,
-                0x22, 0x74, 0x65, 0x73, 0x74, 0x2E, 0x74, 0x78,
-                0x74, 0x22, 0x0D, 0x0A, 0x0D, 0x0A, 0x6d, 0x65,
-                0x73, 0x73, 0x61, 0x67, 0x65,};
-    uint32_t mimemsg_len = sizeof(mimemsg) - 1;
-    TcpSession ssn;
-    Packet *p;
-    ThreadVars th_v;
-    DetectEngineThreadCtx *det_ctx = NULL;
-    Flow f;
-
-    AppLayerParserThreadCtx *alp_tctx = AppLayerParserThreadCtxAlloc();
-
-    memset(&th_v, 0, sizeof(th_v));
-    memset(&f, 0, sizeof(f));
-    memset(&ssn, 0, sizeof(ssn));
-
-    p = UTHBuildPacket(NULL, 0, IPPROTO_TCP);
-
-    FLOW_INITIALIZE(&f);
-    f.protoctx = (void *)&ssn;
-    f.proto = IPPROTO_TCP;
-    f.flags |= FLOW_IPV4;
-    f.alstate = SMTPStateAlloc(NULL, ALPROTO_UNKNOWN);
-
-    MimeDecParseState *state = MimeDecInitParser(&f, NULL);
-    ((MimeDecEntity *)state->stack->top->data)->ctnt_flags = CTNT_IS_ATTACHMENT;
-    state->body_begin = 1;
-
-    FAIL_IF(SMTPProcessDataChunk((uint8_t *)mimemsg, sizeof(mimemsg), state) != 0);
-
-    p->flow = &f;
-    p->flowflags |= FLOW_PKT_TOSERVER;
-    p->flowflags |= FLOW_PKT_ESTABLISHED;
-    p->flags |= PKT_HAS_FLOW|PKT_STREAM_EST|PKT_STREAM_EOF;
-    f.alproto = ALPROTO_SMTP;
-
-    StreamTcpInitConfig(true);
-
-    DetectEngineCtx *de_ctx = DetectEngineCtxInit();
-    FAIL_IF_NULL(de_ctx);
-    de_ctx->flags |= DE_QUIET;
-
-    Signature *s = DetectEngineAppendSig(de_ctx, "alert smtp any any -> any any "
-                                                 "(msg:\"file_data smtp test\"; "
-                                                 "file_data; content:\"message\"; sid:1;)");
-    FAIL_IF_NULL(s);
-
-    SigGroupBuild(de_ctx);
-    DetectEngineThreadCtxInit(&th_v, (void *)de_ctx, (void *)&det_ctx);
-
-    int r = AppLayerParserParse(NULL, alp_tctx, &f, ALPROTO_SMTP,
-            STREAM_TOSERVER | STREAM_START | STREAM_EOF, mimemsg, mimemsg_len);
-    FAIL_IF_NOT(r == 0);
-    FAIL_IF_NULL(f.alstate);
-
-    /* do detect */
-    SigMatchSignatures(&th_v, de_ctx, det_ctx, p);
-    FAIL_IF_NOT(PacketAlertCheck(p, 1));
-
-    AppLayerParserThreadCtxFree(alp_tctx);
-    DetectEngineCtxFree(de_ctx);
-    StreamTcpFreeConfig(true);
-    FLOW_DESTROY(&f);
-    UTHFreePackets(&p, 1);
-    PASS;
-}
-
-static int DetectEngineSMTPFiledataTest02(void)
-{
-    DetectEngineCtx *de_ctx = DetectEngineCtxInit();
-    FAIL_IF_NOT(de_ctx);
-    de_ctx->flags |= DE_QUIET;
-
-    Signature *s = DetectEngineAppendSig(de_ctx, "alert smtp any any -> any any "
-                                                 "(msg:\"file_data smtp test\"; "
-                                                 "file_data; content:\"message\"; sid:1;)");
-    FAIL_IF_NULL(s);
-
-    FAIL_IF_NOT(s->flags & SIG_FLAG_TOSERVER);
-    DetectEngineCtxFree(de_ctx);
-    PASS;
-}
```

Comment:
I think this one is like the cases in detect-uricontent where tests that only parse signatures should not be removed 
(like https://github.com/OISF/suricata/pull/6746/commits/b53da1e5366f1ae4fd5bc70f5f592aa5820ca1c0#r780184577)